rust-runtime/aws-smithy-client/src/lib.rs

```diff
@@ -166,7 +166,10 @@ impl<C, M> Client<C, M> {
 
     /// Set the [`AsyncSleep`] function that the client will use to create things like timeout futures.
     pub fn set_sleep_impl(&mut self, sleep_impl: Option<Arc<dyn AsyncSleep>>) {
-        self.sleep_impl = sleep_impl;
+        self.sleep_impl = sleep_impl.clone();
+        if let Some(sleep_impl) = sleep_impl {
+            self.retry_policy.with_sleep_impl(sleep_impl);
+        }
```

Comment:
Not sure if this would ever be a use-case, but once you set a sleep impl, you wouldn't be able to unset it without recreating the client.